### PR TITLE
CASSANDRA-17543 4.0: Split ReadRepairQueryTypesTest to avoid JUnit timeouts

### PR DESCRIPTION
**Pull request by adelapena**

### CODE REVIEW

File: test/distributed/org/apache/cassandra/distributed/test/ReadRepairCollectionQueriesTest.java [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import static org.apache.cassandra.distributed.shared.AssertUtils.row;
+
+/**
+ * {@link ReadRepairQueryTester} for queries on collections.
+ */
+@RunWith(Parameterized.class)
```
Comment (maedhroz) [inline test/distributed/org/apache/cassandra/distributed/test/ReadRepairCollectionQueriesTest.java]: nit: Do the subclasses need `@RunWith(Parameterized.class)`?

File: test/distributed/org/apache/cassandra/distributed/test/ReadRepairQueryTester.java [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+ *     <li>Run the tested read query again selecting all the columns</li>
+ *     <li>Verify that the previous read has triggered read repair propagating the column deletion</li>
+ *     <li>Delete one of the involved rows in just one node</li>
+ *     <li>Run the tested read query again selecting all the columns</li>
+ *     <li>Verify that the previous read has triggered read repair propagating the row deletions</li>
+ *     <li>Verify the final status of each node and drop the table</li>
+ * </ul>
+ */
+@RunWith(Parameterized.class)
+public abstract class ReadRepairQueryTester extends TestBaseImpl
```
Comment (maedhroz) [inline test/distributed/org/apache/cassandra/distributed/test/ReadRepairQueryTester.java:73]: naming nit: `ReadRepairQueryTester` -> `ReadRepairQueryTest`

File: test/distributed/org/apache/cassandra/distributed/test/ReadRepairQueryTester.java [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+import org.apache.cassandra.distributed.Cluster;
+import org.apache.cassandra.service.reads.repair.ReadRepairStrategy;
+
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static org.apache.cassandra.distributed.shared.AssertUtils.assertEquals;
+import static org.apache.cassandra.distributed.shared.AssertUtils.assertRows;
+import static org.apache.cassandra.service.reads.repair.ReadRepairStrategy.NONE;
+
+/**
+ * Tests basic read repair functionality with different types of query types and schemas.
```
Comment (maedhroz) [inline test/distributed/org/apache/cassandra/distributed/test/ReadRepairQueryTester.java]:
```suggestion
 * Base class for tests around read repair functionality with different query types and schemas.
```

File: test/distributed/org/apache/cassandra/distributed/test/ReadRepairQueryTester.java [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+ * and that unselected rows/columns are not repaired.
+ * <p>
+ * The tests are parameterized for:
+ * <ul>
+ *     <li><Both {@code NONE} and {@code BLOCKING} read repair stratregies/li>
+ *     <li>Data to be repaired residing on the query coordinator or a replica</li>
+ *     <li>Data to be repaired residing on memtables or flushed to sstables</li>
+ * </ul>
+ * <p>
+ * All the included tests have a similar behaviour:
```
Comment (maedhroz) [inline test/distributed/org/apache/cassandra/distributed/test/ReadRepairQueryTester.java]:
```suggestion
 * All derived tests follow a similar pattern:
```

File: test/distributed/org/apache/cassandra/distributed/test/ReadRepairCollectionQueriesTest.java (hunk 1 above)
Comment (adelapena) [reply test/distributed/org/apache/cassandra/distributed/test/ReadRepairCollectionQueriesTest.java]: Not really, removed from subclasses.

File: test/distributed/org/apache/cassandra/distributed/test/ReadRepairQueryTester.java (hunk 2 above)
Comment (adelapena) [reply test/distributed/org/apache/cassandra/distributed/test/ReadRepairQueryTester.java:73]: `ReadRepairQueryTest` produces [an initialization error](https://app.circleci.com/pipelines/github/adelapena/cassandra/1482/workflows/877e7dd4-6e46-4610-aa3b-5a67cc7b2a9d/jobs/15338), I'm keeping `ReadRepairQueryTester` which is in line with other abstract test classes without tests such as `CQLTester`.

**Pull request closed without merging**